### PR TITLE
CR-263 duplicates from limma

### PR DESCRIPTION
**Pull request by EpigeneMax**
## Description
See [Jira card](https://epigenelabsteam.atlassian.net/browse/CR-263)

## Motivation and Context
Splitting the port of `limma` into smaller PR to ease the integration process

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] No code (non-breaking change that does not impact code: formatting, build system, documentation...)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My changes are in a separate branch named after a Jira card, or -- if not applicable -- with a descriptive name.
- [x] I have descriptive commit messages, with a short title (first line).
- [x] I have added and/or updated tests to reflect my changes.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have updated the [Changelog](../CHANGELOG.md).
- [x] My code and documentation follow the [numpy docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).
- [x] I have reviewed possible licensing issues, preferably with a person in charge of the topic.

### CODE REVIEW

File: inmoose/limma/dups.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        The list of genes with duplicate spots removed.
+
+    Raises
+    ------
+    ValueError
+        If ndups is less than or equal to 1.
+    """
+
+    if ndups <= 1:
+        return genelist
```
Comment (Lea-Meunier) [inline inmoose/limma/dups.py:86]: Here if you follow the function doc it should raise an error

File: inmoose/limma/dups.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    spacing : int, optional
+        the number of rows between duplicate measurements for a gene. Defaults to 1.
+
+    Returns
+    -------
+    ndarray
+        the unwrapped matrix with duplicate spots combined.
+    """
+    if ndups == 1:
+        return M
```
Comment (Lea-Meunier) [inline inmoose/limma/dups.py:43]: what happens if `ndups == 0`?  it return an empty ndarray?

**Review by Lea-Meunier — CHANGES_REQUESTED** [review]
I have few comment on this PR, 
and maybe in this PR you can remove the comment of the test that use `ndups` in the file [tests/limma/test_lmFit.py] (row 28)(https://github.com/epigenelabs/inmoose/pull/45/files/bd93c066687a7b2b9cd1dec48101dcf3fe8048b8#diff-e09e5f2a67c6749a9baabc9a09e9fb51761389e409720e8019a218becabb0350) 

```
def test_lmFit(self):
        with self.assertRaisesRegex(
            ValueError, expected_regex="the correlation must be set"
        ):
            lmFit(self.y, self.design, ndups=2)
        # fit = lmFit(self.y, self.design, ndups=2, correlation=0.9)
        # coef_ref = np.array(
        #    [
        #        [1.144165577, 0.03340760],
        #        [-0.212545740, -0.14219338],
        #        [-0.017368967, -0.09863220],
        #        [-0.248713327, 0.04916677],
        #        [0.001386212, 0.02736341],
        #    ]
        # )
        # self.assertTrue(np.allclose(fit.coefficients, coef_ref))```

File: inmoose/limma/dups.py (hunk 2 above)
Comment (EpigeneMax) [reply inmoose/limma/dups.py:45]: fixed

File: inmoose/limma/dups.py (hunk 1 above)
Comment (EpigeneMax) [reply inmoose/limma/dups.py:82]: fixed

**Review by Lea-Meunier — APPROVED** [review]
Thanks for the changes! 
All seams good to me, the only comment left is maybe it is the good PR to un-commented the test in the test_lmFit.py file in `test_lmFit` function with test of the function with `ndups` parameters 
`lmFit(self.y, self.design, ndups=2, correlation=0.9)`

**Pull request merged**